### DISCUSSION TITLE
Incorrect status_code - response returning 'HTTP/1.1 200 OK'

### DISCUSSION THREAD

**Discussion by martin-foka** [dymmond/esmerald#393] [Potential Issue]
Hello,

I am encountering an issue where the response _status_code_ is always returned as `HTTP/1.1 200 OK`, despite other status codes being specified and expected.

## Steps to Reproduce

1. Create a simple Esmerald app with the following code:

```python
from esmerald import Esmerald, Gateway, get, Response
from lilya import status

@get()
def welcome() -> Response:
    return Response({"message": "Hello!"}, status_code=status.HTTP_202_ACCEPTED)

app = Esmerald(
    routes=[
        Gateway("/", handler=welcome),
    ]
)
```

2. Run the application and hit the endpoint to observe the HTTP response status code

**Expected Behavior:** The response should return a `HTTP/1.1 202 Accepted` status code as specified in the response.
**Actual Behavior:** The response status code is `HTTP/1.1 200 OK`, regardless of the specified status code.

### Additional Information
This issue was reproduced on the following setups:
```
Operating Systems: Ubuntu 22.04
Python Version: 3.11
Esmerald Version: 3.3.7
Uvicorn Version: 0.30.6
```
```
Operating Systems:  Windows 11
Python Version: 3.11 and 3.12
Esmerald Version: 3.3.7
Uvicorn Version: 0.30.6
```
Please let me know if you need any further information.

## Questions

Is there something I might be overlooking? Could this be related to my specific environment setup, or is it a broader issue?

When I comment out the following lines of code in the Esmerald source, the correct status codes are returned:
[esmerald/routing/base.py#L297-L298](https://github.com/dymmond/esmerald/blob/main/esmerald/routing/base.py#L297-L298)

**Comment by tarsil** [comment]
@maRT-sk hmm! Weird. Let us investigate this. Next week we have a release planned and if this is an issue after investigation, it will also be fixed there. 

But before anything else, can you do this:

@get(status_code=status. HTTP_202_ACCEPTED)
def welcome() -> Response:
    return Response({"message": "Hello!"})

And let us know if the error still persists, please?

**Reply by martin-foka** [reply]
No, I am receiving the expected status code when it is used within the decorator, so it functions correctly in that context.
However, I find the approach from my snippet to be more useful.

**Reply by tarsil** [reply]
Yes. Work the decorator is expected to work as you confirmed. The reason for this is because Esmerald is smart enough to see your return annotation and build internally the response object. 

We can although also consider adding your approach. What you can do is also looking an Lilya https://www.lilya.dev/responses/?h=make_response#custom-encoders-and-the-make_response

Make response. Esmerald is built on top of Lilya.

We will consider your suggestion for the next release coming out next week 👍🏼 

I hope you are also enjoying Esmerald.

**Reply by tarsil** [reply]
@maRT-sk this will be coming in the next release, the possibility of overriding the response from the object itself. The decorators still have the defaults but you will now be able to override from the object itself.

EDIT: https://www.esmerald.dev/release-notes/

It was released

**Comment by tarsil** [comment]
Addressed
